diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GenerateAutolinkingNewArchitecturesFileTask.kt b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GenerateAutolinkingNewArchitecturesFileTask.kt
index 4c36c8a..e87a58f 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GenerateAutolinkingNewArchitecturesFileTask.kt
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GenerateAutolinkingNewArchitecturesFileTask.kt
@@ -92,6 +92,7 @@ abstract class GenerateAutolinkingNewArchitecturesFileTask : DefaultTask() {
       packages: List<ModelAutolinkingDependenciesPlatformAndroidJson>
   ): String {
     val packagesWithLibraryNames = packages.filter { android -> android.libraryName != null }
+    val packagesWithCxxModules = packages.filter { android -> android.cxxModuleHeaderName != null }
 
     val cppIncludes =
         packagesWithLibraryNames.joinToString("\n") { dep ->
@@ -100,11 +101,11 @@ abstract class GenerateAutolinkingNewArchitecturesFileTask : DefaultTask() {
             include +=
                 "\n#include <${COMPONENT_INCLUDE_PATH}/${dep.libraryName}/${COMPONENT_DESCRIPTOR_FILENAME}>"
           }
-          if (dep.cxxModuleHeaderName != null) {
-            include += "\n#include <${dep.cxxModuleHeaderName}.h>"
-          }
           include
-        }
+        } +
+            packagesWithCxxModules.joinToString("", prefix = "\n") { dep ->
+              "#include <${dep.cxxModuleHeaderName}.h>\n"
+            }
 
     val cppTurboModuleJavaProviders =
         packagesWithLibraryNames.joinToString("\n") { dep ->
@@ -120,8 +121,7 @@ abstract class GenerateAutolinkingNewArchitecturesFileTask : DefaultTask() {
         }
 
     val cppTurboModuleCxxProviders =
-        packagesWithLibraryNames
-            .filter { it.cxxModuleHeaderName != null }
+        packagesWithCxxModules
             .joinToString("\n") { dep ->
               val cxxModuleHeaderName = dep.cxxModuleHeaderName
               // language=cpp
